did-tezos/src/lib.rs

```diff
@@ -60,20 +69,82 @@ impl DIDResolver for DIDTz {
             ..Default::default()
         };
 
-        let doc = Document {
-            context: Contexts::One(Context::URI(DEFAULT_CONTEXT.to_string())),
-            id: did.to_string(),
-            authentication: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
-            assertion_method: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
-            verification_method: Some(vec![VerificationMethod::Map(VerificationMethodMap {
-                id: String::from(vm_didurl),
-                type_: proof_type.to_string(),
-                controller: did.to_string(),
-                blockchain_account_id: Some(format!("{}@tezos:{}", address.to_string(), network)),
-                ..Default::default()
-            })]),
-            ..Default::default()
+        let mut doc = DIDTz::tier1_derivation(did, &vm_didurl, proof_type, &address, &network);
+
+        let service = match DIDTz::tier2_resolution(did, &address, &network).await {
+            Ok(s) => s,
+            Err(e) => {
+                return (
+                    ResolutionMetadata {
+                        error: Some(e.to_string()),
+                        ..Default::default()
+                    },
+                    Some(doc),
+                    None,
+                )
+            }
         };
+        doc.service = Some(vec![service]);
```

Comment:
I get an error now when trying to resolve any of the example DIDs:
```
$ didkit did-resolve did:tz:tz1TzrmTBSuiVHV2VfMnGRMYvTEPCP42oSM8
DID Manager not found.
```
Should resolution proceed if no manager found?

Also, offline use won't be possible if connection to the network is required. If offline use is needed, maybe there could be a resolution input metadata option to skip tier 2.

Reply:
i think that did-tezos will simply require online use for consistency. we can use other did methods if offline use is desired

Reply:
At the moment, the specs state
> Success can only be achieved after processing all resolution layers. 

We could add such option, I don't know what @wyc thinks?

Reply:
i think that!

Reply:
Is it still supposed to be possible to use `did:tz` without publishing anything to the blockchain? In this case wouldn't the query for DID Manager return no result, but resolution should still succeed?

Reply:
Yes, that's correct.

Reply:
Thanks again for your insights @clehner, looks like this has been resolved in https://github.com/spruceid/ssi/pull/98/commits/ee45e53d6c39800907ee5ed286ca278cad0de5dc

Reply:
Works, thanks